standard/tier/tier_data.lua

```diff
@@ -0,0 +1,110 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Tier/Data
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+return {
+	tiers = {
+		{
+			value = '1',
+			sort = 'A1',
+			name = 'S-Tier',
+			short = 'S',
+			link = 'S-Tier Tournaments',
+			category = 'S-Tier Tournaments',
+		},
+		{
+			value = '2',
+			sort = 'A2',
+			name = 'A-Tier',
+			short = 'A',
+			link = 'A-Tier Tournaments',
+			category = 'A-Tier Tournaments',
+		},
+		{
+			value = '3',
+			sort = 'A3',
+			name = 'B-Tier',
+			short = 'B',
+			link = 'B-Tier Tournaments',
+			category = 'B-Tier Tournaments',
+		},
+		{
+			value = '4',
+			sort = 'A4',
+			name = 'C-Tier',
+			short = 'C',
+			link = 'C-Tier Tournaments',
+			category = 'C-Tier Tournaments',
+		},
+		{
+			value = '5',
+			sort = 'A5',
+			name = 'D-Tier',
+			short = 'D',
+			link = 'D-Tier Tournaments',
+			category = 'D-Tier Tournaments',
+		},
+		[''] = {
+			value = nil,
+			sort = 'B2',
+			name = 'Undefined',
+			short = '?',
+		},
+	},
+
+	tierTypes = {
+		monthly = {
+			value = 'Monthly',
+			sort = 'A6',
+			name = 'Monthly',
+			short = 'Mon.',
+			link = 'Monthly Tournaments',
+			category = 'Monthly Tournaments',
+		},
+		weekly = {
+			value = 'Weekly',
+			sort = 'A7',
+			name = 'Weekly',
+			short = 'Week.',
+			link = 'Weekly Tournaments',
+			category = 'Weekly Tournaments',
+		},
+		qualifier = {
+			value = 'Qualifier',
+			sort = 'A8',
+			name = 'Qualifier',
+			short = 'Qual.',
+			link = 'Qualifier Tournaments',
+			category = 'Qualifier Tournaments',
+		},
+		misc = {
+			value = 'Misc',
+			sort = 'A9',
+			name = 'Misc',
+			short = 'Misc',
+			link = 'Miscellaneous Tournaments',
+			category = 'Miscellaneous Tournaments',
+		},
+		showmatch = {
+			value = 'Showmatch',
+			sort = 'B1',
+			name = 'Showmatch',
+			short = 'Showm.',
+			link = 'Showmatches',
+			category = 'Showmatch Tournaments',
+		},
+	},
+
+	-- only for legacy conversion reasons
+	tierToNumber = {
+		['s-tier'] = 1,
+		['a-tier'] = 2,
+		['b-tier'] = 3,
+		['c-tier'] = 4,
+		['d-tier'] = 5,
+	}
```

Comment:
Maybe a function in the module instead (memorized table)

Reply:
something like this?
```lua
Tier.legacyNumbers = FnUtil.memoize(function()
	return Table.map(TierData.tiers, function(key, data) return data.name:lower(), tonumber(key) or '' end)
end)

--- Legacy: Converts legacy tier input to its numeric value. DEPRECATED!!!
---@param tier string|integer|nil
---@return integer
---@deprecated
function Tier.toNumber(tier)
	return Tier.legacyNumbers[tier]
end
```

Reply:
yeah